lib/experimental/class-wp-navigation-fallbacks-gutenberg.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * WP_Navigation_Fallbacks_Gutenberg class
+ *
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @package Gutenberg
+ * @subpackage Navigation
+ * @since 6.3.0
+ */
+
+require __DIR__ . '/class-wp-nav-menu-conversion-gutenberg.php';
+
+/**
+ * Manages fallback behavior for Navigation menus.
+ *
+ * @access public
+ */
+class WP_Navigation_Fallbacks_Gutenberg {
```

Comment:
Why are all the methods here static? What is the benefit of this?

Reply:
I solicited advice on this aspect [here](https://github.com/WordPress/gutenberg/pull/48698#discussion_r1150987995).

Reply:
Static are bad practice. Example of article about it. https://r.je/static-methods-bad-practice

Static method bring little benefit in code. Either it should be a function or a method. 
I prefer code as an object as it is better to test, but I dont care either way.

Reply:
@hellofromtonya All the things you are lists there are just benefits of OOP and does not really explain why it needs to be a static. 

If you want to call a method on an object, then you can just do thing. 

```
$data = ( new WP_Navigation_Fallback_Gutenberg() )->get_fallback();
```

I don't see why it has be to static methods. I do agree that new functionality that goes into core should for the most part be built into classes. What I don't like is the need for the methods and properties to be static. Static properties can "cache" values and make testing harder.

Reply:
>Static properties can "cache" values and make testing harder.
Testing is a tiny bit harder but easily achievable by adding a property reset in the `tear_down()`. If the property is non-public, gain access via Reflection.

>All the things you are lists there are just benefits of OOP and does not really explain why it needs to be a static.
This class is not an object though. It is not describing an object. Rather, it's a series of grouped functionality. I don't think shoehorning non-objects into OOP is the right choice either.

And @azaozz further explains the why and benefits for Core https://github.com/WordPress/gutenberg/pull/48698#issuecomment-1517129204.